swagger.yaml

```diff
@@ -0,0 +1,334 @@
+    required: true
+    description: "The unique filter ID for customising a dataset"
+    in: path
+  dimension_name:
+    name: dimension_name
+    type: string
+    required: true
+    description: "The name of a dimension"
+    in: path
+  value:
+    name: value
+    type: string
+    required: true
+    description: "The single value for a dimension"
+    in: path
+  dimension:
+    name: dimension
+    schema:
+      $ref: '#/definitions/Dimension'
+    required: true
+    description: "A dimension to filter the dataset"
+    in: body
+  filter_job:
+    name: filter_job
+    schema:
+      $ref: '#/definitions/NewJob'
+    required: true
+    description: "Model of all editable properties within a filter job"
+    in: body
+paths:
+  /filters:
+    post:
+      tags:
+      - "Filter"
+      summary: "Create a filter job for a dataset"
+      description: "Create a job so that dimensions can be added to filter a dataset"
+      produces:
+      - "application/json"
+      parameters:
+      - $ref: '#/parameters/filter_job'
+      responses:
+        200:
+          description: "Job was created"
+          schema:
+            $ref: '#/definitions/JobState'
+        500:
+          $ref: '#/responses/InternalError'
+  /filters/{filter_id}:
+    parameters:
+      - $ref: '#/parameters/filter_id'
+    get:
+      tags:
+      - "Filter"
+      summary: "Get a description of a filter job"
+      description: "Get the state of the filter job"
+      produces:
+      - "application/json"
+      responses:
+        200:
+          description: "The job was found and a `JobState` is returned"
+          schema:
+            $ref: '#/definitions/JobState'
+        404:
+           $ref: '#/responses/FilterJobNotFound'
+        500:
+          $ref: '#/responses/InternalError'
+    put:
+      tags:
+      - "Filter"
+      summary: "Update the state of the job"
+      description: |
+        Update the filter job by providing new properties
+      parameters:
+      - $ref: '#/parameters/filter_job'
+      responses:
+        200:
+          description: "The state has been updated"
+        400:
+          description: "The job has been locked as it has been `submitted` to be processed"
+        404:
+          $ref: '#/responses/FilterJobNotFound'
+        500:
+          $ref: '#/responses/InternalError'
+  /filters/{filter_id}/dimensions:
+    get:
+      tags:
+       - "Filter"
+      summary: "Get all dimensions used by this filter"
+      description: |
+        Return a list of all dimensions which are going to be used to filter on
+      parameters:
+      - $ref: '#/parameters/filter_id'
+      responses:
+        200:
+          description: "A list of dimension URLs"
+          schema:
+            type: array
+            items:
+              type: string
+        404:
+          $ref: '#/responses/FilterJobNotFound'
+        500:
+          $ref: '#/responses/InternalError'
+  /filters/{filter_id}/dimensions/{dimension_name}:
```

Comment:
Good spot, this also makes the uri smaller and stops the word dimension being repeated 👍